### ISSUE TITLE
Inconsistent loss function between the code and the publication

### ISSUE THREAD

**Issue by Voytech-0** [eamid/trimap#28]
In the [code](https://github.com/eamid/trimap/blob/f6542cf1e91dabd472ec1e489afa9bcb8091296c/trimap/trimap_.py#L464C9-L464C55), the loss is implemented using `loss += weights[t] * 1.0 / (1.0 + d_ik / d_ij)`, indicating a loss function of $\ell_{i j k}:=\omega_{i j k}\\frac{s(y_{i},y_{j})}{s(y_{i},y_{j})+s(y_{i},y_{k})}$.

 However, in the paper TriMap: Large-scale Dimensionality Reduction Using Triplets, the loss is defined as $\ell_{i j k}:=\omega_{i j k}\\frac{s(y_{i},y_{k})}{s(y_{i},y_{j})+s(y_{i},y_{k})}$.

Empirically, the current implementation in the code performs much better and is also consistent with the statement made in the paper that 

> The loss of the triplet (i,j,k) approaches zero as d_ij decreases and d_ik increases.
